### ISSUE TITLE
[BUG] schema mappings are not applied to `parameters` schemas

### ISSUE THREAD

**Issue by Gama11** [OpenAPITools/openapi-generator#12885]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

Schema mappings don't seem to be applied to schemas referenced from an endpoint's `parameters`. With this:

```
openapi-generator-cli generate -g java -i example.yml -o output --schema-mappings=Limit=com.example.Limit
```

the generated code still uses a raw `Integer` rather than `com.example.Limit`:

```java
public List<Pet> listPets(Integer limit) throws ApiException {
```

In contrast, `--schema-mappings=Pet=com.example.Pet` works as expected:

```java
public List<com.example.Pet> listPets(Integer limit) throws ApiException {
```

##### openapi-generator version

6.0.1

##### OpenAPI declaration file content or url

This is a modified version of the pet store example, with the limit query parameter extracted into a separate schema:

```yml
openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
  license:
    name: MIT
servers:
  - url: http://petstore.swagger.io/v1
paths:
  /pets:
    get:
      summary: List all pets
      operationId: listPets
      tags:
        - pets
      parameters:
        - name: limit
          in: query
          description: How many items to return at one time (max 100)
          required: false
          schema:
            $ref: "#/components/schemas/Limit"
      responses:
        '200':
          description: A paged array of pets
          content:
            application/json:    
              schema:
                $ref: "#/components/schemas/Pets"
components:
  schemas:
    Limit:
      type: integer
      format: int32
    Pet:
      type: object
      required:
        - id
        - name
      properties:
        id:
          type: integer
          format: int64
        name:
          type: string
        tag:
          type: string
    Pets:
      type: array
      items:
        $ref: "#/components/schemas/Pet"
```

##### Related issues/PRs

#12600

**Comment by dpozinen** [comment]
Looks like a symptom of this issue. With the schema below I am getting different outputs pre and post 6.0.0
```
paths:
  /admin/applications:
    post:
      operationId: "createApplication"
      parameters:
        - $ref: "#/components/parameters/Meta-Caller-Name"
...
components:
  parameters:
    Meta-Caller-Name:
      name: Meta-Caller-Name
      in: header
      description: "Service name that call api"
      required: true
      schema:
        enum: [ swagger ]
        example: swagger
        type: string
```
Notice that `@Schema` is not generated after the updated.
```
    // BEFORE 6.0.0
    @RequestMapping(...)
    default ResponseEntity<SaveApplicationResponse> createApplication(
        @Parameter(name = "Meta-Caller-Name", schema = @Schema(description = "", allowableValues = { "swagger" })) String metaCallerName
    ) {
        return getDelegate().createApplication(metaCallerName);
    }

    // AFTER 6.0.0
    @RequestMapping(...)
    default ResponseEntity<SaveApplicationResponse> createApplication(
        @Parameter(name = "Meta-Caller-Name") String metaCallerName
    ) {
        return getDelegate().createApplication(metaCallerName);
    }
```
This impacts for example `allowableValues` which are used by swagger to generate `examples` on the UI.

**Comment by wing328** [comment]
can you please try the latest master? there were fixes related to how schema mapping works in the parameters.

**Comment by Gama11** [comment]
It looks like this is indeed fixed with 6.1.0.
